### PR TITLE
Release preparations

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Checking if all release machinery is ready.

**Test plan**

All should be green.

### CODE REVIEW

**Comment by XVilka** [comment]
So only missing things are:
- [x] 2 failing tests for Alpine static build: https://github.com/rizinorg/rizin/issues/350
- [ ] Unpublished [rz-pipe.py](https://github.com/rizinorg/rz-pipe) on [PyPI](https://pypi.org/).

**Comment by ret2libc** [comment]
The pipe2 error is because that still uses the acr/make build system. But honestly... I would not release a DEB that way...  @thestr4ng3r  ?

AFAIK there is also https://man7.org/linux/man-pages/man7/debhelper.7.html that has integration with meson/ninja so we should be able to do something with that, but I haven't looked at it yet. I suggest we drop  the .deb file for now. In any case, distros hopefully will package that for us at some point :)

**Comment by XVilka** [comment]
@ret2libc there is **zero** documentation on how to do this, so the old way could work for now. It's not an error, just a warning.

**Comment by thestr4ng3r** [comment]
Imo we should not release debs like this at all.

**Comment by XVilka** [comment]
I remember some people use `*.deb` files. Ideally, we should build those binaries statically, with Alpine and musl, and pack then into the deb, so they will work almost anywhere without the 3rd party dependencies. But that's a task for the 0.2.0 release instead.

**Comment by XVilka** [comment]
Regarding porting building the Debian package we can see https://github.com/linuxmint/xed/tree/master/debian as an example.

**Comment by ret2libc** [comment]
I agree with @thestr4ng3r on not releasing .deb files this way. I'm honestly not sure they will work on various debian/ubuntu versions :(

**Comment by XVilka** [comment]
Removed `*.deb` packages building and uploading then.

**Comment by ret2libc** [comment]
I think the build dir used to run `meson dist` should be done with `enable_tests=false`. We probably do not want to ship rz-test as part of the release. Also, We should probably find a way to remove from the release tarball all the test/db stuff?

File: librz/util/sys.c [hunk 1]
```diff
@@ -1378,6 +1378,9 @@ RZ_API void rz_sys_info_free(RSysInfo *si) {
 
 #if __UNIX__ && HAVE_PIPE2
 #include <fcntl.h>
+#ifndef _GNU_SOURCE
```
Comment (ret2libc) [inline librz/util/sys.c]: this is not necessary. It is already defined in meson.build

Comment (XVilka) [reply librz/util/sys.c]: It was necessary for the ACR-based `sys/debian.sh`. But because we removed the use of it, I will remove those lines too.

**Comment by thestr4ng3r** [comment]
> Also, We should probably find a way to remove from the release tarball all the test/db stuff?

I think you should be able to run all tests from a release tarball too. The main question there would be about the testbins, but the makefile (acr/buildsystem-independent) in there can fetch it.

**Comment by XVilka** [comment]
Interesting bug on Alpine that isn't related to the debugging:
```
[XX] db/tools/rz_ax rz-ax -t "1234567890 GMT-1"
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '!rz-ax -t "1234567890 GMT-1" | cut -d " " -f 1,2' -
-- stdout
@@ -1 +1 @@
-Fri Feb
+Fri May
```

**Comment by XVilka** [comment]
@ret2libc @thestr4ng3r it's ready now, please check & test the artifacts/etc

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -375,27 +375,34 @@ jobs:
       working-directory: rizin
     - name: Compile with meson
       run: |
-        CFLAGS="-static" LDFLAGS="-static" meson --prefix=${HOME}/.local --buildtype release --default-library static build
+        mkdir -p ${HOME}/rizin-static
+        CFLAGS="-static" LDFLAGS="-static" meson --prefix=${HOME}/rizin-static --buildtype release --default-library static build
         ninja -C build && ninja -C build install
+        tar -C ${HOME}/rizin-static --strip-components=3 --xz -cf rizin-static.tar.xz ${HOME}/rizin-static
```
Comment (ret2libc) [inline .github/workflows/ci.yml]: why this? isn't the created tarball already in good shape?

Comment (XVilka) [reply .github/workflows/ci.yml]: It isn't

Comment (XVilka) [reply .github/workflows/ci.yml]: Doesn't help anyway

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -793,41 +720,14 @@ jobs:
...
+    - name: Upload rizin static x86_64 tar.gz
       uses: actions/upload-release-asset@v1
       env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       with:
           upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./rizin-dev-ubuntu-1804-deb/rizin-dev_amd64.deb
-          asset_name: rizin-dev-ubuntu-1804_${{ steps.extract_version.outputs.branch }}_amd64.deb
+          asset_path: ./rizin-static/rizin-static.tar.xz
+          asset_name: rizin-${{ steps.extract_version.outputs.branch }}-static-x86_64.tar.gz
```
Comment (ret2libc) [inline .github/workflows/ci.yml]: why the conversion from tar.xz to tar.gz?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -375,27 +375,34 @@ jobs:
...
       uses: actions/upload-artifact@v2
       with:
         name: test-results-static
-        path: test/results.json
+        path: rizin/test/results.json
+    - name: Upload static build
+      uses: actions/upload-artifact@v2
+      with:
+        name: rizin-static.tar.xz
+        path: rizin/rizin-static.tar.xz
```
Comment (ret2libc) [inline .github/workflows/ci.yml:403]: if i try to download the file, I find another archive inside this one. I think there's something wrong.

Comment (XVilka) [reply .github/workflows/ci.yml:403]: It's GitHub failure - it always creates ZIP files.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (XVilka) [reply .github/workflows/ci.yml]: No conversion, forgot to change.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (XVilka) [reply .github/workflows/ci.yml:403]: See https://github.com/actions/upload-artifact/issues/39

**Review by ret2libc — APPROVED** [review]

**Pull request merged**